crates/state/src/values/color.rs

```diff
@@ -13,29 +15,40 @@ pub trait DisplayColor {
 }
 
 impl Parse for Color {
-    fn parse(value: &str) -> Result<Self, ParseError> {
-        match value {
-            "red" => Ok(Color::RED),
-            "green" => Ok(Color::GREEN),
-            "blue" => Ok(Color::BLUE),
-            "yellow" => Ok(Color::YELLOW),
-            "black" => Ok(Color::BLACK),
-            "gray" => Ok(Color::GRAY),
-            "white" => Ok(Color::WHITE),
-            "orange" => Ok(Color::from_rgb(255, 165, 0)),
-            "transparent" => Ok(Color::TRANSPARENT),
-            _ => {
-                if value.starts_with("hsl(") {
-                    parse_hsl(value)
-                } else if value.starts_with("rgb(") {
-                    parse_rgb(value)
-                } else if value.starts_with('#') {
-                    parse_hex_color(value)
-                } else {
-                    Err(ParseError)
-                }
-            }
-        }
+    fn parse(parser: &mut Parser) -> Result<Self, ParseError> {
+        parser
+            .consume_one_of(&[
+                Token::Pound,
+                Token::ident("rgb"),
+                Token::ident("hsl"),
+                Token::ident("red"),
+                Token::ident("green"),
+                Token::ident("blue"),
+                Token::ident("yellow"),
+                Token::ident("black"),
+                Token::ident("gray"),
+                Token::ident("white"),
+                Token::ident("orange"),
+                Token::ident("transparent"),
```

Comment:
Is it possible to not have to specify all these values twice? Here and below in the match

Reply:
because `and_then` will be called only if the result is `Ok`, and `consume_one_of` will return `Err` if the current token is not among `values`

Reply:
Whoops, I mixed up the comments

Reply:
Regarding specifying all these values twice - I don't like it myself, but otherwise `&mut parser` can't be used for `parser_hex_color`, `parser_gb` and `parse_hsl` (because the parser will already be borrowed as mutable when `consume_map` is called)

Reply:
Ah, I figured out how to solve that problem.